docs/reference-guides/block-api/block-bindings.md

```diff
@@ -232,6 +234,138 @@ The `setValues` function updates all the values of the source of the block bound
 - `dispatch` returns an `object` of the store's action creators. [More about dispatch](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#dispatch).
 - `select` returns an `object` of a given store's selectors. [More info in their docs.](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#select).
 
+#### editorUI
+
+_**Note:** Since WordPress 6.9._
+
+The `editorUI` property provides configuration for how the source appears in the block editor's bindings UI. It should be a function that returns a configuration object, allowing for dynamic data based on context.
+
+When `editorUI` is a function, it receives an `object` as an argument with the following properties:
+
+- `context` returns an `object` of the current block context, defined in the `usesContext` property.
+- `select` returns an `object` of a given store's selectors.
+
+The function or object must return/be an object with this structure:
+
+- `mode`: `string` - Either `'dropdown'` for a dropdown interface or `'modal'` for a modal interface.
+- `data`: `array` - An array of available fields, each with:
+    - `key`: `string` - Unique identifier for the field
+    - `label`: `string` - Human-readable label
+    - `value`: `any` - Current value of the field
+    - `type`: `string` - Data type (`'string'`, `'number'`, `'boolean'`, etc.)
+- `getArgs`: `function` (optional). Accepts an object with an item. - Returns the source arguments for the selected item. Used for dropdown mode to customize the binding arguments.
+- `isSelected`: `function` (optional). Accepts an object with item and binding. - Returns whether the field is currently selected. Used for dropdown mode to determine the checked state.
```

Comment:
This needs updating to reflect the latest changes. (Same for the examples below.)